pkg/yurtmanager/controller/yurtappset/workloadmanager/tweaks.go

```diff
@@ -42,7 +42,8 @@ func GetNodePoolTweaksFromYurtAppSet(cli client.Client, nodepoolName string, yas
 	for _, yasTweak := range yas.Spec.Workload.WorkloadTweaks {
 		if isNodePoolRelated(&np, yasTweak.Pools, yasTweak.NodePoolSelector) {
 			klog.V(4).Infof("nodepool %s is related to yurtappset %s/%s, add tweaks", nodepoolName, yas.Namespace, yas.Name)
-			tweaksList = append(tweaksList, &yasTweak.Tweaks)
```

Comment:
Would you like to explain why last Tweak will be returned here?

Reply:
The address of the yasTweak variable remains unchanged within the for loop. Since tweaksList is actually a pointer slice, append operates on the same address, resulting in all elements pointing to the last one by the end of the for loop.

Reply:
Got it. Thank you for your kindly reply.

Reply:
> The address of the yasTweak variable remains unchanged within the for loop. Since tweaksList is actually a pointer slice, append operates on the same address, resulting in all elements pointing to the last one by the end of the for loop.

This fix may be unnecessary after go1.22.
https://go.dev/blog/loopvar-preview